src/lightning/pytorch/strategies/launchers/multiprocessing.py

```diff
@@ -182,6 +191,8 @@ def _recover_results_in_main_process(self, worker_output: "_WorkerOutput", train
 
         trainer.state = worker_output.trainer_state
 
+        self._already_fit |= trainer.state.fn == TrainerFn.FITTING
```

Comment:
could we move this to the end of the launch() method, so that both the check and the update are in the same method. I don't think it fits here well.